frontend/src/components/dashboard/aliases/AliasList.test.tsx

```diff
@@ -0,0 +1,92 @@
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { mockConfigModule } from "../../../../__mocks__/configMock";
+import { getMockRandomAlias } from "../../../../__mocks__/hooks/api/aliases";
+import { getMockProfileData } from "../../../../__mocks__/hooks/api/profile";
+import {
+  getReturnValueWhenDisabled as getLocalLabelsWhenDisabled,
+  getReturnValueWhenEnabled as getLocalLabelsWhenEnabled,
+  setMockLocalLabels,
+  setMockLocalLabelsOnce,
+} from "../../../../__mocks__/hooks/localLabels";
+import { mockFluentReact } from "../../../../__mocks__/modules/fluent__react";
+import { AliasList } from "./AliasList";
+
+jest.mock("@fluent/react", () => mockFluentReact);
+jest.mock("../../../config.ts", () => mockConfigModule);
+setMockLocalLabels();
+
+describe("<AliasList>", () => {
+  it("sends a request to the back-end to update the label if server-side label storage is enabled", async () => {
+    const updateCallback = jest.fn();
+    const storeLocalLabelCallback = jest.fn();
+    setMockLocalLabelsOnce(getLocalLabelsWhenDisabled(storeLocalLabelCallback));
```

Comment:
There is a LOT of redirection going on here, including some `import ... as` which made this harder to trace. The result of all this logic is ... `setMockLabelsOnce(getReturnValueWhenDisabled as getLocalLabelsWhenDisabled => [null, jest.fn()]` ? Is all that redirection necessary?

question: Also this test is for "server-side label storage is enabled" but it's using `getLocalLabelsWhenDisabled`?

Reply:
Yeah, so far I've been going with names that make sense in a module's own context, but might be duplicated across modules, and thus renaming them at the call-site (similar: every component with its own `Props` type, rather than e.g. `WhatsNewContentProps`), and especially in the case of mocks, the function `getReturnValue*` by itself doesn't really make sense outside of the module's context. But maybe a pattern that might be easier to follow here is to alias the entire module, and then simply call the functions from that? I applied that in 9cd3b02c, which feels like an improvement.

What makes this extra confusing is that there's different, but similar, things that can be enabled or disabled - this test verifies behaviour if server-side label storage is enabled, which automatically means that _local_ storage of labels is _disabled_, hence the use of that function. Hopefully that also makes more sense with the module-wide alias?

(Thought: If you prefer that pattern as well, I'd like to see if we could find a lint rule that prevents aliasing of specific imports.)

Reply:
Hrm ... yeah there's just a lot going on here I suppose. I'm not sure I can come up with anything better either.